torch/distributed/device_mesh.py

```diff
@@ -444,34 +446,78 @@ def __init__(
 
             # Skip process group initialization if xla device or init backend is False
             # TODO(yeounoh) implement DeviceMesh backend and register XLA backend.
-            if device_type != "xla":
-                # always try to create default (world) pg, even if it is not initialized
-                # already. The world pg is used for device mesh identity (rank) on each
-                # process (we need to know if the current global rank is in the mesh or not).
-                if _init_backend:
-                    self._get_or_create_default_group()
-                    self._init_process_groups()
-
-                if is_initialized() and get_backend() == "threaded":
-                    self._thread_id = threading.get_ident()
-
-                # calculate the coordinates of the current global rank on the mesh
-                rank_coords = (self.mesh == get_rank()).nonzero()
-                assert rank_coords.size(0) in (0, 1)
-                self._coordinate_on_dim: Optional[List[int]] = (
-                    rank_coords[0].tolist() if rank_coords.size(0) > 0 else None
-                )
-
-        def _get_or_create_default_group(self):
-            default_initialized = is_initialized()
-            if not default_initialized:
-                init_process_group()
+            if device_type == "xla":
+                # TODO: I (kwen2501) don't know why xla also skips the
+                # coordinate calculation below. Just keep things as is for now.
+                return
+
+            # We may opportunistically create default (world) pg, if it is not
+            # initialized already. (Take it as a convenience feature that may be
+            # revoke if we have to).
+            # The world pg is used for device mesh identity (rank) on each
+            # process (we need to know if the current global rank is in the mesh
+            # or not).
+            if _init_backend:
+                self._get_or_create_default_group()
+                # Create the subgroups for each dimension of the mesh.
+                self._init_process_groups()
+
+            # Must have been initialized by this point
+            if get_backend() == "threaded":
```

Comment:
for my education, what is backend == "threaded"?

Reply:
I think it is the backend name for MTPG. I just left this line here in case it is needed for some purpose (e.g. for testing using MTPG, I guess).

Reply:
> for my education, what is backend == "threaded"?

Ye. This is for MTPG. Specifically, for MTPG, we will need to include the `_thread_id` in the DeviceMesh's hash to avoid hash collision on different threads.